DuggaSys/sectioned.js

```diff
@@ -2230,16 +2232,32 @@ function drawSwimlanes() {
     "<linearGradient gradientUnits='userSpaceOnUse' x1='0' x2='300' y1='0' y2='0' id='fadeTextRed'>" +
     "<stop offset='85%' stop-opacity='1' stop-color='#FF0000' /><stop offset='100%' stop-opacity='0'/> </linearGradient></defs>";
 
+  const storedTheme = localStorage.getItem('themeBlack');
+
   for (var i = 0; i < weekLength; i++) {
     str += "<rect x='" + (i * weekwidth) + "' y='" + (15) + "' width='" +
       (weekwidth) + "' height='" + (weekheight * (deadlineEntries.length + 1)) + "' ";
-    if ((i % 2) == 0) {
+    // The following code decides the color scheme of the course timeline 
+    // (The svg showing how far into a course we are and assignment deadlines)
+    if ((i % 2) == 0 && !(storedTheme.includes("blackTheme.css"))) { // All even columns should be light gray in white theme
       str += "fill='#ededed' />";
-    } else {
+    } else if((i % 2) != 0 && !(storedTheme.includes("blackTheme.css"))){ // All uneven columns should be white in white theme
       str += "fill='#ffffff' />";
+    } else if((i % 2) == 0 && (storedTheme.includes("blackTheme.css"))){ // All even columns should be dark gray in black theme
+      str += "fill='#121212' />";
+    }else{ // All uneven columns should be a lighter gray in white theme
+      str += "fill='#434343' />";
     }
-    str += `<text x='${((i * weekwidth) + (weekwidth * 0.5))}' y='${(33)}'
-    font-family='Arial' font-size='12px' fill='black' text-anchor='middle'>${(i + 1)}</text>`;
+
+    //Changes text colors of week numbers in course SVG (course schedule with deadlines)
+    if(storedTheme.includes("blackTheme.css")){ //If theme is black then week numbers in the course schedule should be white
+      str += `<text x='${((i * weekwidth) + (weekwidth * 0.5))}' y='${(33)}'
+      font-family='Arial' font-size='12px' fill='white' text-anchor='middle'>${(i + 1)}</text>`;
+    }else{ //If theme is white then week numbers in the course schedule should be black
+      str += `<text x='${((i * weekwidth) + (weekwidth * 0.5))}' y='${(33)}'
+      font-family='Arial' font-size='12px' fill='black' text-anchor='middle'>${(i + 1)}</text>`;
+    }
+
```

Comment:
I think this solution may be over complicating things. It should be possible to just add one class to the parent of all swimlane/schedule elements. And then use css selectors to set the style. For example, striped rows could be added in CSS via [css nthchild odd](https://www.w3schools.com/howto/howto_css_table_zebra.asp).